src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -0,0 +1,116 @@
+   * server, not when running in a test.
+   */
+
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(userStore.getInstance());
+    setConversationStore(conversationStore.getInstance());
+    setMessageStore(messageStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for
+   * use by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+
+      String username = (String)request.getSession().getAttribute("user");
+      if (username == null) {
+        // user is not logged in, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      User user = userStore.getUser(username);
+
+      if (user == null) {
+        // user was not found, don't let them access the admin page
+        response.sendRedirect("/");
+        return;
+      }
+      adminList.add("admin");
+      if (user.getIsAdmin() == true) {
+        if (!adminList.contains(user.getName())){
+          adminList.add(user.getName());
+        }
+      }
+
+      request.setAttribute("adminList", adminList);
+      if (adminList.contains(user.getName())) {
+      /* an attempt to grab information from the stores to display on the page
+      if the user is admin*/
+          List<User> userList = UserStore.getInstance().getAllUsers();
+          int numOfUsers = userList.size();
+          int numOfConvos = conversationStore.getInstance().getAllConversations().size();
+          int numOfMessages = messageStore.getInstance().getAllMessages().size();
+          String newestUser = userList.get(userList.size()-1).getName();
+          String mostActiveUser = "";
+          for(int i =0; i< numOfUsers; i++){
+            UUID userId = userList.get(i).getId();
+              int mostMessages=0;
+            if (messageStore.getMessagesFromAuthor(userId).size() > mostMessages){
+              mostMessages = messageStore.getMessagesFromAuthor(userId).size();
+              mostActiveUser = userStore.getInstance().getUser(userId).getName();
```

Comment:
How will private messages affect this? Say that User A has 50 messages, all private, and User B has 20 messages, all public. Do we want people to know User A is the most active?

Reply:
This will probably need a separate change if we think private messages shouldn't count. I'd open an issue to track it separately.

Reply:
Once the private messages are implemented I can change the stats to show the amount of public and private messages as well total messages. I can from then make most active user based on public messages if we store the two types in different list.